src/main/java/teammates/ui/controller/StudentCourseJoinAuthenticatedAction.java

```diff
@@ -30,20 +30,7 @@
     @Override
     protected ActionResult execute() throws EntityDoesNotExistException {
         Assumption.assertNotNull(regkey);
-        String nextUrl;
-        if (regkey.contains("${amp}" + Const.ParamsNames.NEXT_URL + "=")) {
-            /*
-             * Here regkey may contain the nextUrl as well. This is due to
-             * a workaround which replaces "&" with a placeholder "${amp}", thus the
-             * next parameter, nextUrl, is treated as part of the "regkey".
-             * TODO move this process to Action class is possible.
-             */
-            String[] split = regkey.split("\\$\\{amp\\}" + Const.ParamsNames.NEXT_URL + "=");
-            regkey = split[0];
-            nextUrl = split[1];
-        } else {
-            nextUrl = getRequestParamValue(Const.ParamsNames.NEXT_URL);
-        }
+        String nextUrl = nextUrlFromRegkey == null ? getRequestParamValue(Const.ParamsNames.NEXT_URL) : nextUrlFromRegkey;
```

Comment:
I think comments should be added here too to explain why this is being done. Otherwise someone looking at this file may be lost. You could link to `parseAndInitializeRegkeyFromRequest` in the comment.